openedx/features/ondemand_email_preferences/tests/test_views.py

```diff
@@ -0,0 +1,50 @@
+"""
+All tests for ondemand preferences views.
+"""
+import mock
+from django.test import Client
+from django.urls import reverse
+from rest_framework import status
+
+from lms.djangoapps.onboarding.tests.factories import UserFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+class OnDemandEmailPreferencesViews(ModuleStoreTestCase):
+    """
+    This class contains all tests for ondemand preferences views.
+    """
+
+    def setUp(self):
+        super(OnDemandEmailPreferencesViews, self).setUp()
+        self.user = UserFactory(password='password')
+        self.client = Client()
+        self.client.login(username=self.user.username, password='password')
+        self.course = CourseFactory.create(display_name='test course', run='Testing_course')
+        self.update_email_preferences_comp_path = reverse('update_email_preferences_component',
+                                                          args=(str(self.course.id),))
+
+    def test_unauthorization_redirection_to_get_email_preferences(self):
```

Comment:
I think the method name should be `test_unauthorization_redirection_on_get_email_preferences` since we are not redirected to get_email_preferences, rather we are redirecting to the sign in page upon landing on get_email_preferences.